File: src/main/java/org/goodgoodgood/freetubeexservice/domain/music/PlaylistRepository.java
package org.goodgoodgood.freetubeexservice.domain.music;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Arrays;
import java.util.List;

public interface PlaylistRepository extends JpaRepository<Playlist, Long> {

    @Query("select p from Playlist p where p.user.email = :email")
    List<Playlist> findAllByUserEmail(String email);
}

File: kotlin-api/src/main/kotlin/org/goodgoodgood/freetubeexservice/domain/music/repository/PlaylistRepository.kt
package org.goodgoodgood.freetubeexservice.domain.music.repository

import org.goodgoodgood.freetubeexservice.domain.music.Playlist
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface PlaylistRepository: JpaRepository<Playlist, Long> {
    @Query("select p from Playlist p where p.user.email = :email")
    fun findAllByUserEmail(email: String?): List<Playlist>
}